standard/types.md

```diff
@@ -568,7 +568,7 @@ The exact definition of the generic type `Expression<TDelegate>` as well as the
 
 Two things are important to make explicit:
 
-- Not all lambda expressions can be converted to expression trees. For instance, lambda expressions with statement bodies, and lambda expressions containing assignment expressions cannot be represented. In these cases, a conversion still exists, but it will fail at compile-time. These exceptions are detailed in [§11.7.3](conversions.md#1173-evaluation-of-anonymous-function-conversions-to-expression-tree-types)).
+- Not all lambda expressions can be converted to expression trees. For instance, lambda expressions with statement bodies, and lambda expressions containing assignment expressions cannot be represented. In these cases, a conversion still exists, but it will fail at compile-time.
```

Comment:
Starting back at line 556 (I can't attach a comment there) it states that expression trees, an implementation defined type, "are efficient" – maybe "shall be", "must be", "are required to be", but surely not "are"? It carries on to state that they make "make the structure of the lambda expression transparent and explicit" – again this is an implementation defined type, so maybe "shall make…", "must make…" etc but not simply "and make".

In this section we also see repetition of material from §11.7 without at least x-refs, and maybe less repetition is possible.

I think this is proving a hard feature to Standardise because it is essentially implementation defined and the Standard is trying to be implementation neutral but hasn't much to work with, while also implementation specific material (e.g. it being efficient above) has crept in.

Reply:
Agree that we shouldn't have efficient in there - removed.

I've reworked this section a bit, restoring the cross-reference (but moving it) and removing duplication. I've retained the part about the `Compile` method, but it's no longer in a list.

Reply:
Also removed the bit about return types and parameters which seemed to be unnecessary.